docs/dev/components/operation.md

```diff
@@ -0,0 +1,81 @@
+# Operation
+
+The [operation module](https://github.com/gchq/Gaffer/tree/master/core/operation) contains the `Operation` interfaces and core operation implementations.
+
+It is assumed that all Gaffer graphs will be able to handle these core operations.
+
+An `Operation` implementation defines an operation to be processed on a graph, or on a set of results which are returned by another operation. An `Operation` class contains the configuration required to tell Gaffer how to carry out the operation. For example, the `AddElements` operation contains the elements to be added.
+The `GetElements` operation contains the seeds to use to find elements in the graph and the filters to apply to the query.
+The operation classes themselves should not contain the logic to carry out the operation (as this may vary between the different supported store types), just the configuration.
+
+For each operation, each Gaffer store will have an `OperationHandler`, where the processing logic is contained. This enables operations to be handled differently in each store.
+
+Operations can be chained together to form an `OperationChain`. When an operation chain is executed on a Gaffer graph the output of one operation is passed to the input of the next.
+
+An `OperationChain.Builder` is provided to help with constructing a valid operation chain - it ensures the output type of an operation matches the input type of the next.
+
+## How to write an Operation
+
+Operations should be written to be as generic as possible to allow them to be applied to different graphs/stores.
+
+Operations must be JSON serialisable in order to be used via the REST API - i.e. there must be a public constructor and all the fields should have getters and setters.
+
+Operation implementations need to implement the `Operation` interface and the extra interfaces they wish to make use of. For example an operation that takes a single input value should implement the `Input` interface.
+
+Here is a list of some of the common interfaces:
+- uk.gov.gchq.gaffer.operation.io.Input
```

Comment:
Should these be in back ticks as they're kind of code/class names e.g. `uk.gov.gchq.gaffer.operation.io.Input`

Reply:
Would probably be good to have that.